### ISSUE TITLE
Add missing JSDoc methods

### ISSUE THREAD

**Issue by dsherret** [dsherret/ts-morph#753]
There are some missing JSDoc methods as seen in [this report](https://github.com/dsherret/ts-morph/blob/latest/packages/ts-morph/wrapped-nodes.md).

Summary:

* JSDocAugmentsTag
    * `class`
* JSDocSignature
    * `typeParameters`
    * `parameters`
* JSDocTypedefTag
    * `fullName`
    * `typeExpression`

This would be an easy issue for someone to implement if they feel up for it.

**Comment by JvPy** [comment]
still open? would like to contribute

**Comment by dsherret** [comment]
@JvPy for sure! That would be great!

I think this would be just for adding the helper methods around inspecting the nodes and the tests for that (no manipulation for now).

Example for implementing `JSDocThisTag` (these steps might be slightly wrong, but will at least help you get going):

1. Add a new `JSDocThisTag` class in *src/compiler/ast/doc* folder that inherits from `JSDocTag`.
2. Export it from the *index.ts* file in that folder.
3. Update `kindToWrapperMappings` to have that `SyntaxKind.JSDocThisTag` pointing at that class.
4. Run `yarn code-generate`
5. Add the helper methods to `JSDocThisTag` for `typeExpression` on that node.
6. Add some basic tests to make sure that method works.

**Comment by dsherret** [comment]
By the way, even if you only want to do a few of these that's great! It might be a lot of work to add them all. Maybe to start just submit a PR for one of the classes to ensure you're on the right track?

**Comment by JvPy** [comment]
> By the way, even if you only want to do a few of these that's great! It might be a lot of work to add them all. Maybe to start just submit a PR for one of the classes or for adding some missing methods to ensure you're on the right track?

I agree. I will do JSDocAuthorTag. I can do more after that, if needed
